src/MSALWrapper/AuthFlow/Web.cs

```diff
@@ -0,0 +1,183 @@
+                {
+                    try
+                    {
+                        var tokenResult = await TaskExecutor.CompleteWithin(
+                            this.logger,
+                            this.silentAuthTimeout,
+                            "Get Token Silent",
+                            (cancellationToken) => this.pcaWrapper.GetTokenSilentAsync(this.scopes, account, cancellationToken),
+                            this.errors)
+                            .ConfigureAwait(false);
+                        tokenResult.SetAuthenticationType(AuthType.Silent);
+
+                        return new AuthFlowResult(tokenResult, this.errors);
+                    }
+                    catch (MsalUiRequiredException ex)
+                    {
+                        this.errors.Add(ex);
+                        this.logger.LogDebug($"Silent auth failed, re-auth is required.\n{ex.Message}");
+                        var tokenResult = await TaskExecutor.CompleteWithin(
+                            this.logger,
+                            this.interactiveAuthTimeout,
+                            "Interactive Auth",
+                            (cancellationToken) => this.pcaWrapper
+                            .WithPromptHint(this.promptHint)
+                            .GetTokenInteractiveAsync(this.scopes, account, cancellationToken),
+                            this.errors)
+                            .ConfigureAwait(false);
+                        tokenResult.SetAuthenticationType(AuthType.Interactive);
+
+                        return new AuthFlowResult(tokenResult, this.errors);
+                    }
+                }
+                catch (MsalUiRequiredException ex)
+                {
+                    this.errors.Add(ex);
+                    this.logger.LogDebug($"Silent auth failed, re-auth is required.\n{ex.Message}");
+                    var tokenResult = await TaskExecutor.CompleteWithin(
+                        this.logger,
+                        this.interactiveAuthTimeout,
+                        "Interactive Auth (with extra claims)",
+                        (cancellationToken) => this.pcaWrapper
+                        .WithPromptHint(this.promptHint)
+                        .GetTokenInteractiveAsync(this.scopes, ex.Claims, cancellationToken),
+                        this.errors)
+                        .ConfigureAwait(false);
+                    tokenResult.SetAuthenticationType(AuthType.Interactive);
+
+                    return new AuthFlowResult(tokenResult, this.errors);
+                }
+            }
+            catch (MsalServiceException ex)
+            {
+                this.logger.LogWarning($"MSAL Service Exception! (Not expected)\n{ex.Message}");
+                this.errors.Add(ex);
+            }
+            catch (MsalClientException ex)
+            {
+                this.logger.LogWarning($"Msal Client Exception! (Not expected)\n{ex.Message}");

```

Comment:
```suggestion
                this.logger.LogWarning($"MSAL Client Exception! (Not expected)\n{ex.Message}");

```

Reply:
There are some other similar occurrences as well. You may want to replace those to `MSAL`.

Reply:
This code was ported from `TokenFetcherPublicClient` which uses MSAL and Msal both. Since this is a nit, I will update it.